app/assets/javascripts/discourse/app/components/topic-map/private-message-map.gjs

```diff
@@ -0,0 +1,257 @@
+          @removeAllowedUser={{@removeAllowedUser}}
+        />
+      {{/each}}
+    </div>
+
+    {{#if this.canShowControls}}
+      <div class="controls">
+        <DButton
+          @action={{this.actionAllowed}}
+          @label={{this.actionAllowedLabel}}
+          class="btn btn-default add-remove-participant-btn"
+        />
+
+        {{#if (and this.canInvite this.isEditing)}}
+          <DButton
+            @action={{@showInvite}}
+            @icon="plus"
+            class="btn btn-default no-text btn-icon add-participant-btn"
+          />
+        {{/if}}
+      </div>
+    {{/if}}
+  </template>
+}
+
+class PmMapUserGroup extends Component {
+  get canRemoveLink() {
+    return this.args.isEditing && this.args.canRemoveAllowedUsers;
+  }
+
+  get groupUrl() {
+    return groupPath(this.args.model.name);
+  }
+
+  <template>
+    <div class="user group">
+      <a href={{this.groupUrl}} class="group-link">
+        {{dIcon "users"}}
+        <span class="group-name">{{@model.name}}</span>
+      </a>
+    </div>
+    {{#if this.canRemoveLink}}
+      <PmRemoveGroupLink
+        @model={{@model}}
+        @removeAllowedGroup={{@removeAllowedGroup}}
+      />
+    {{/if}}
+  </template>
+}
+
+class PmRemoveGroupLink extends Component {
+  @service dialog;
+
+  @action
+  showConfirmDialog() {
+    this.dialog.deleteConfirm({
+      message: I18n.t("private_message_info.remove_allowed_group", {
+        name: this.args.model.name,
+      }),
+      confirmButtonLabel: "private_message_info.remove_group",
+      didConfirm: () => this.args.removeAllowedGroup(this.args.model),
+    });
+  }
+
+  <template>
+    {{! template-lint-disable no-invalid-interactive }}
+    <a
+      class="remove-invited no-text btn-icon btn"
+      {{on "click" this.showConfirmDialog}}
+    >
+      {{dIcon "times"}}
+    </a>
+  </template>
+}
+
+class PmMapUser extends Component {
+  @service site;
+
+  get linkClass() {
+    if (this.site.mobileView) {
+      return "";
+    }
+    return "user-link";
+  }
+
+  get userUrl() {
+    return this.args.model.path;
+  }
+
+  get avatarImage() {
+    return htmlSafe(
+      tinyAvatar(this.args.model.avatar_template, {
+        title: this.args.model.name || this.args.model.username,
+      })
+    );
+  }
+
+  get isCurrentUser() {
+    return this.args.canRemoveSelfId === this.args.model.id;
+  }
+
+  get canRemoveLink() {
+    return (
+      this.args.isEditing &&
+      (this.args.canRemoveAllowedUsers || this.isCurrentUser)
+    );
+  }
+
+  <template>
+    <div class="user">
+      <a class={{this.linkClass}} href={{this.userUrl}}>
+        {{#if this.site.mobileView}}
+          {{this.avatarImage}}
+        {{else}}
+          <a
+            class="trigger-user-card"
+            data-user-card={{@model.username}}
+            title={{@model.username}}
+            aria-hidden="true"
+          >
+            {{this.avatarImage}}
+          </a>
```

Comment:
I'm wondering... is there any reason we *shouldn't* link the user card on mobile?

A long time ago, we didn't support the user card on mobile at all, so that might explain why this old logic existed. I think if it works on mobile, we should remove the `{{#if` and keep the logic consistent.